qiskit/dagcircuit/_dagcircuit.py

```diff
@@ -1341,6 +1352,30 @@ def property_summary(self):
                    "operations": self.count_ops()}
         return summary
 
+    def add_dag_at_the_end(self, dag, layout=None):
```

Comment:
So how is this method different from `compose()`?

Reply:
compose needs the two dags to have the same registers. This wraps compose to run `add_*reg` ahead.